koku/api/provider/serializers.py

```diff
@@ -326,8 +326,9 @@ def create(self, validated_data):
             Provider.objects.filter(authentication=auth).filter(billing_source=bill).filter(customer=customer).count()
         )
         if unique_count != 0:
-            error = {"Error": "A Provider already exists with that Authentication and Billing Source"}
-            raise serializers.ValidationError(error)
+            key = "Error"
+            message = "A Provider already exists with that Authentication and Billing Source"
```

Comment:
@spank50 These are a few source error message strings that were missed as part of our user facing error string cleanup.  That work is captured here: https://docs.google.com/spreadsheets/d/1BgweFAhU3RAU30Bj1rzYQcVxwOBWS1WUBYyjlvD-bFY/edit#gid=0

The `Improved error message` column is what Dayle, Natalie and I landed on.  This might give you a sense of how these error messages are worded.

Do you have any suggestions on what the text should be?  The situation that needs to be covered is when a source (provider) already exists in cost management with the same account configuration.  Additionally we shouldn't be using the Provider name for user facing strings.  It should be Source.

Let me know if you have any questions on this.  Thanks!

Reply:
@boaz0 It would be great if we could internally log the source that already existed and then make a more general user facing string that would be the language that we use from @spank50 's guidance.

Reply:
sure thing

Reply:
@dccurtis done :+1: 
waiting on @spank50 response.

Reply:
Based on the examples provided in the spreadsheet, I suggest the following wording. I do have some questions first.
- Is this an AWS error, Azure error, or both?
- The authentication/billing source is edited in AWS/Azure, and the user knows how to edit and remove the duplicate?
- Is "Authentication and Billing Source" one thing, or is it two things, "Authentication Source and Billing Source"? 
If one thing:
"Cost management does not allow a duplicate source for authentication and billing. Edit your [AWS/Azure] source and remove the duplicate information."

Reply:
Thanks for the feedback @spank50!  I should have given you more background before, but here we go!

This error can occur for all providers (AWS, Azure, and OCP).  The terms `authentication` and `billing_source` are generic internal names that help define an account.

Each provider type has a different definition of what `authentication` and `billing_source` means:

**AWS**
Authentication: roleARN
Billing Source: S3 Bucket

**Azure**
Authentication: Client ID, Tenant ID, and Subscription ID 
Billing Source: Storage Resource Group, Storage Account

**OCP**
Authentication: Cluster ID
Billing Source: N/A

A duplicate source would happen if:
1. The user entered identical values for both "authentication" and "billing source" for the respective source type.
2. The name of the source would be different than the one that already exists.  This is because platform sources guarantees that source names are globally unique.

When the user creates a source they are prompted to enter the specific source values (roleARN, s3 bucket, client id, etc) and not authentication or billing source (internal names).

The user facing source error strings are generic error strings at this point in time and they don't provide too many specifics as to what the problem actually is.  I'm definitely open to enhancing this in the future, but today they are generic error messages.

Now for this case I think there is a lot of value in telling the user what pre-existing source already has the account settings since then they would be able to take action on fixing the problem (removing the old one, edit the old one, bail on creating the new one, etc).

@boaz0 made some changes to allow for us to log the existing source name.  I'm thinking we could use this in the user facing string.

With all this being said.  What do you think about something like:

"Cost management does not allow duplicate accounts. [Source Name] already exists.  Edit source settings to configure a new source."

@nlcwong might be interested this duplicate source situation.

Reply:
Thanks for the explanation @dccurtis. Your suggested wording looks good to me.

Reply:
Done :+1: